ntcore/src/main/java/edu/wpi/first/networktables/NetworkTableType.java

```diff
@@ -48,10 +51,84 @@ public static NetworkTableType getFromInt(int value) {
         return kDoubleArray;
       case 0x40:
         return kStringArray;
-      case 0x80:
-        return kRpc;
+      case 0x100:
+        return kInteger;
+      case 0x200:
+        return kFloat;
+      case 0x400:
+        return kIntegerArray;
+      case 0x800:
+        return kFloatArray;
       default:
         return kUnassigned;
     }
   }
+
+  /**
+   * Convert from a type string to an enum type.
+   *
+   * @param typeString type string
+   * @return The kind
+   */
+  public static NetworkTableType getFromString(String typeString) {
+    if ("boolean".equals(typeString)) {
+      return kBoolean;
+    } else if ("double".equals(typeString)) {
+      return kDouble;
+    } else if ("float".equals(typeString)) {
+      return kFloat;
+    } else if ("int".equals(typeString)) {
+      return kInteger;
+    } else if ("string".equals(typeString) || "json".equals(typeString)) {
+      return kString;
+    } else if ("boolean[]".equals(typeString)) {
+      return kBooleanArray;
+    } else if ("double[]".equals(typeString)) {
+      return kDoubleArray;
+    } else if ("float[]".equals(typeString)) {
+      return kFloatArray;
+    } else if ("int[]".equals(typeString)) {
+      return kIntegerArray;
+    } else if ("string[]".equals(typeString)) {
+      return kStringArray;
+    } else if ("".equals(typeString)) {
+      return kUnassigned;
+    } else {
+      return kRaw;
+    }
+  }
+
+  /**
+   * Gets string from generic data value.
+   *
+   * @param data the data to check
+   * @return type string of the data, or empty string if no match
+   */
+  public static String getStringFromObject(Object data) {
+    if (data instanceof Boolean) {
```

Comment:
Can `instanceof` be used in a switch? Again, strings as constants somewhere (maybe as a member in the enums?) would be great

Reply:
No, instanceof can't be used in a switch.

Reply:
Added constants to both places.